### PR TITLE
Delete spaces created on PRs after a certain amount of time

### LINKED ISSUE #2245: Delete PR Spaces automatically
- [x] I have searched to see if a similar issue already exists.

We should delete the spaces created automatically during PR reviews after the PR is merged or after some given amount of time has passed after the PR has been merged. Keeping around the spaces post merge for a bit may help keep a history of the demos in case we realize something is broken.

### PR DESCRIPTION
**Pull request by freddyaboulton**
# Description

Fixes #2245

# Checklist:

- [x] I have performed a self-review of my own code
- [x] My code follows the style guidelines of this project
- [x] I have commented my code in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: scripts/delete_old_spaces.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+def delete_space(space_id: str, hf_token: str, api_client: Optional[HfApi] = None):
+    api_client = api_client or HfApi()
+    api_client.delete_repo(repo_id=space_id, token=hf_token, repo_type="space")
+
+
+def get_spaces_to_delete(
+    n_days: int, org_name: str, api_client: Optional[HfApi] = None
+):
+    api_client = api_client or HfApi()
```
Comment (abidlabs) [inline scripts/delete_old_spaces.py:16]: Out of curiosity, why pass in an `api_client` as a parameter like this? We can just define a single instance of `HfApi()` outside of the function?

**Review by abidlabs — APPROVED** [review]

**Comment by abidlabs** [comment]
LGTM @freddyaboulton! I haven't tested the code, but I did go through and inspect it and it looks ✅

File: scripts/delete_old_spaces.py (hunk 1 above)
Comment (freddyaboulton) [reply scripts/delete_old_spaces.py:16]: I think it makes the function more general in the event we want to reuse them in other scripts in the future. We do currently define a single instance of api_client and pass it to all the functions.

**Comment by freddyaboulton** [comment]
Thanks for the review @abidlabs !

**Pull request merged**